Code/GraphMol/TautomerQuery/TautomerQuery.h

```diff
@@ -91,7 +104,54 @@ class RDKIT_TAUTOMERQUERY_EXPORT TautomerQuery {
 
   const std::vector<size_t> getModifiedBonds() const { return d_modifiedBonds; }
 
+  //! serializes (pickles) to a stream
+  void toStream(std::ostream &ss) const;
+  //! returns a string with a serialized (pickled) representation
+  std::string serialize() const;
+  //! initializes from a stream pickle
+  void initFromStream(std::istream &ss);
+  //! initializes from a string pickle
+  void initFromString(const std::string &text);
+
   friend class TautomerQueryMatcher;
+
+#ifdef RDK_USE_BOOST_SERIALIZATION
+  template <class Archive>
+  void save(Archive &ar, const unsigned int version) const {
+    RDUNUSED_PARAM(version);
+    std::vector<std::string> pkls;
+    for (const auto taut : d_tautomers) {
+      std::string pkl;
+      MolPickler::pickleMol(*taut, pkl);
+      pkls.push_back(pkl);
+    }
+    ar &pkls;
+    std::string molpkl;
+    MolPickler::pickleMol(*d_templateMolecule, molpkl);
+    ar &molpkl;
+    ar &d_modifiedAtoms;
+    ar &d_modifiedBonds;
+  }
+
+  template <class Archive>
+  void load(Archive &ar, const unsigned int version) {
+    RDUNUSED_PARAM(version);
+
+    std::vector<std::string> pkls;
+    ar &pkls;
```

Comment:
I had to look up this notation in the boost docs.  I think it would be clearer to use the more familiar stream notation of `ar >> plks` instead of `ar &pkls`